module/VuFind/src/VuFind/ILS/Driver/CombinedDriver.php

```diff
@@ -0,0 +1,1346 @@
+     */
+    public function getStatus($id)
+    {
+        return $this->combineArraysOfAssociativeArrays('getStatus', [$id]);
+    }
+
+    /**
+     * Get Statuses
+     *
+     * This is responsible for retrieving the status information for a
+     * collection of records.
+     *
+     * @param array $ids The array of record ids to retrieve the status for
+     *
+     * @throws ILSException
+     * @return array     An array of getStatus() return values on success.
+     */
+    public function getStatuses($ids)
+    {
+        return $this->combineMultipleArraysOfAssociativeArrays('getStatuses', [$ids], 'id');
+    }
+
+    /**
+     * Get suppressed authority records
+     *
+     * @return array ID numbers of suppressed authority records in the system.
+     */
+    public function getSuppressedAuthorityRecords()
+    {
+        return $this->defaultCall('getSuppressedAuthorityRecords', func_get_args());
+    }
+
+    /**
+     * Get suppressed records.
+     *
+     * @throws ILSException
+     * @return array ID numbers of suppressed records in the system.
+     */
+    public function getSuppressedRecords()
+    {
+        return $this->defaultCall('getSuppressedRecords', func_get_args());
+    }
+
+    /**
+     * Provide an array of URL data (in the same format returned by the record
+     * driver's getURLs method) for the specified bibliographic record.
+     *
+     * @param string $id Bibliographic record ID
+     *
+     * @return array
+     */
+    public function getUrlsForRecord(string $id): array
+    {
+        return $this->defaultCall('getUrlsForRecord', func_get_args());
+    }
+
+    /**
+     * Has Holdings
+     *
+     * This is responsible for determining if holdings exist for a particular
+     * bibliographic id
+     *
+     * @param string $id The record id to retrieve the holdings for
+     *
+     * @return bool True if holdings exist, False if they do not
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function hasHoldings($id)
+    {
+        return $this->defaultCall('hasHoldings', func_get_args());
+    }
+
+    /**
+     * Get Hidden Login Mode
+     *
+     * This is responsible for indicating whether login should be hidden.
+     *
+     * @return bool true if the login should be hidden, false if not
+     */
+    public function loginIsHidden()
+    {
+        return $this->defaultCall('loginIsHidden', func_get_args());
+    }
+
+    /**
+     * Patron Login
+     *
+     * This is responsible for authenticating a patron against the catalog.
+     *
+     * @param string $username The patron barcode
+     * @param string $password The patron password
+     *
+     * @throws ILSException
+     * @return mixed           Associative array of patron info on successful login,
+     * null on unsuccessful login.
+     */
+    public function patronLogin($username, $password)
+    {
+        return $this->defaultCall('patronLogin', func_get_args());
```

Comment:
Default return value should be null.

Reply:
Now, an exception is thrown if the method is not supported.